Sources/BSON/BSONObjectID.swift

```diff
@@ -26,7 +29,22 @@ public struct BSONObjectID: Equatable, Hashable, CustomStringConvertible {
     ///   - `BSONError.InvalidArgumentError` if string passed is not a valid BSONObjectID
     /// - SeeAlso: https://github.com/mongodb/specifications/blob/master/source/objectid.rst
     public init(_ hex: String) throws {
-        fatalError("ah")
+        guard hex.utf8.count == (BSONObjectID.LENGTH * 2) else {
+            throw BSONError.InvalidArgumentError(message: "Cannot create ObjectId from \(hex)")
```

Comment:
since we know the issue here (invalid length) maybe we could say that in the error message?